sdk/eventhub/Azure.Messaging.EventHubs/README.md

```diff
@@ -115,20 +115,76 @@ await using (var client = new EventHubClient(connectionString, eventHubName))
+
+To consume events for all partitions of an Event Hub, you'll create an `EventProcessor` for a specific consumer group.  When an Event Hub is created, it provides a default consumer group that can be used to get started.  
+
+The `EventProcessor` will delegate processing of events to a `IPartitionProcessor` implementation that you provide, allowing your logic to focus on the logic needed to provide value while the processor holds responsibility for managing the underlying consumer operations.  In our example, we will focus on building the `EventProcessor` and use a very minimal partition processor that does no actual processing.
+
+```csharp
+public class SimplePartitionProcessor : IPartitionProcessor
+{
+    public Task InitializeAsync() => Task.CompletedTask;
+    public Task CloseAsync((PartitionProcessorCloseReason reason) => Task.CompletedTask;
+    public Task ProcessEventsAsync(IEnumerable<EventData> events, CancellationToken cancellationToken) => Task.CompletedTask;
+    public Task ProcessErrorAsync(Exception exception, CancellationToken cancellationToken) => Task.CompletedTask;
+}
+
+// This function is used by the EventProcessor to create a new partition processor for each partition of the 
+// Event Hub.  
+//
+// Its implementation may take various forms, such as a local function (as illustrated), a class or 
+// static method, an inline lambda, or a variable of type Func<PartitionContext, CheckpointManager, IPartitionProcessor>.
+
+IPartitionProcessor SimpleProcessorFactory(PartitionContext context, CheckpointManager manager)
+{
+    return new SimplePartitionProcessor();
+}
+
+var connectionString = "<< CONNECTION STRING FOR THE EVENT HUBS NAMESPACE >>";
+var eventHubName = "<< NAME OF THE EVENT HUB >>";
+
+await using (var client = new EventHubClient(connectionString, eventHubName))
+{
+    string consumerGroup = EventHubConsumer.DefaultConsumerGroup;
+    PartitionManager partitionManager = new InMemoryPartitionManager(Console.WriteLine);
+    
+    EventProcessor processor = new EventProcessor(consumerGroup, client, SimpleProcessorFactory);
+    await processor.StartAsync();
+    
+    // At this point, the processor is consuming events from each partition of the Event Hub and
+    // delegating them to the SimplePartitionProcessor instance created for that partition.  This
+    // processing takes place in the background and will not block. 
+    //
+    // It is important to note that the processor does not own the EventHubClient that was passed into it.  
+    // You are responsible for ensuring that it is disposed after processing.  It is also important that it
+    // not be closed or disposed during the time that the EventProcessor is running.  
+    //
+    // In this example, we'll stop processing after five minutes.
+    
+    await Task.Delay(TimeSpan.FromMinutes(5));
+    await processor.StopAsync();
+}
+```
 ## Troubleshooting
 
 ### Common exceptions
 
-#### Operation Cancelled
+#### Operation Canceled
```

Comment:
Sentence casing for headers? Would this be "Operation cancelled"? ;) See what I did there.

Reply:
I hate this with every fiber of my being, and came very close to asserting that this is a development artifact and not one that falls under the documentation standards _(which I truly believe.)_    However, you're right and I've changed it.